src/actions/getSaleStatus.ts

```diff
@@ -1,26 +1,32 @@
 import * as ethers from "ethers";
-import { WolfSale } from "../contracts/types";
+import { AirWild2Sale } from "../contracts/types";
 
 import { SaleStatus } from "../types";
 
-export const getSaleStatus = async (contract: WolfSale) => {
+export const getSaleStatus = async (contract: AirWild2Sale) => {
   const saleStarted = await contract.saleStarted();
 
   if (!saleStarted) return SaleStatus.NotStarted;
 
   const currentBlock = await contract.provider.getBlockNumber();
   const startBlock = await contract.saleStartBlock();
-  const duration = await contract.privateSaleDuration();
+  const duration = await contract.mintlistDurations(
+    await contract.currentMerkleRootIndex()
+  );
 
   if (ethers.BigNumber.from(currentBlock).gt(startBlock.add(duration))) {
-    const numberSold = await contract.domainsSold();
-    const forSaleInPhase = await contract.numberForSaleForCurrentPhase();
+    const currentMintlist = await contract.currentMerkleRootIndex();
 
-    if (numberSold.gte(forSaleInPhase)) {
-      return SaleStatus.Ended;
+    if (currentMintlist.toNumber() >= 1) {
+      return SaleStatus.PublicSale;
     }
 
-    return SaleStatus.PublicSale;
+    const numSold = await contract.domainsSold();
+    const totalForSale = await contract.totalForSale();
+
```

Comment:
What happens if the sale time elapses? Technically the sale would have ended and it would be impossible to purchase, but the SDK will still be returning `PublicSale` when it's actually `Ended`